blog/content/post/2024-03-15-uefi-sb-implementation-mpl.md

```diff
@@ -0,0 +1,385 @@
+abstract: "This post explains how we tackled the problem of implementing Secure Boot
+           on MPL's PIP platforms. The process included testing the platform's
+           compatibility with Secure Boot and integrating automatic image
+           signing into an existing Yocto layer."
+cover: /covers/mpl-pip44.jpg
+author: pawel.langowski
+layout: post
+published: true
+date: 2024-03-15
+archives: "2024"
+
+tags:
+  - secure boot
+  - uefi
+  - mpl
+  - yocto
+categories:
+  - Firmware
+  - Security
+
+---
+
+MPL is a Swiss company, which designs and manufactures embedded computers and
+microcontroller hardware for rugged environment, extended temperature range,
+and with long-term availability. The PIP series is a family of low-power,
+read-to-use embedded computers manufactured by MPL. Recently we tackled the
+problem of implementing UEFI Secure Boot on platforms from the PIP4x series. The
+goal of the project was to verify that the platform in question is compatible
+with Secure Boot and to enable automatic signing of system components during
+build in Yocto.
+
+## Verifying Secure Boot compatibility
+
+The first step we took was verifying that Secure Boot could indeed be
+implemented on the platform and identifying potential issues and vulnerabilities.
+To do that, we developed and executed various automated tests within the Dasharo
+OSFV
+([Open Source Firmware Validation](https://github.com/Dasharo/open-source-firmware-validation))
+environment. It utilizes the [Robot Framework](https://robotframework.org/) – an
+open-source automation framework, which simplifies the creation and execution
+of test cases.
+
+### Setup
+
+In order to prepare the host machine for Dasharo OSFV infrastructure, follow
+these steps:
+
+- Install the following packages:
+
+   ```bash
+   sudo apt-get install git virtualenv python3-pip
+   ```
+
+- Clone the OFSV repository and update linked submodules
+
+  ```bash
+  git clone https://github.com/Dasharo/open-source-firmware-validation
+  cd open-source-firmware-validation
+  git submodule update --init --checkout
+  ```
+
+- Prepare virtualenv
+
+  ```bash
+  python3 -m virtualenv venv
+  source venv/bin/activate
+  pip install -r requirements.txt
+  ```
+
+If you want, you can run the tests in QEMU. To set it up, follow these steps:
+
+- Navigate to `scripts/ci` in the OSFV repository.
+- Create a `qemu-data` directory and create a disk `qcow` file there.
+
+  ```bash
+  mkdir scripts/ci
+  qemu-img create -f qcow2 qemu-data/hdd.qcow 20G
+  ```
+
+- Download the Ubuntu installer and run QEMU with the `os-installer` argument
+
+  ```bash
+  INSTALLER_PATH=path/to/installer ./qemu-run.sh graphic os-installer
+  ```
+
+- Go through the installation and close QEMU.
+
+QEMU is now ready for OSFV. You run it by executing the following command:
+
+```bash
+./qemu-run.sh graphic os-installer
+```
+
+### Test implementation
+
+Development of test cases in the Robot Framework consists of defining
```

Comment:
please provide link to RF

Reply:
I provide the link in line 41, where i first mention RF.